google/detectors/rce/CVE-2017-5638/src/main/java/com/google/tsunami/plugins/detectors/rce/CVE-2017-5638/ApacheStrutsContentTypeRceDetector.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.rce.cve_2017_5638;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects Apache Struts Command Injection via Content-Type header
+ * (CVE-2017-5638).
+ *
+ * <p>A vulnerable server will use an invalid Content-Type header value as an OGNL expression which
+ * is able to execute system commands under the privileges of the web server. We test it by
+ * acccessing the HttpServletResponse and adding a custom header with a random value. If the header
+ * is reflected in the response, then we know our code was executed.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "ApacheStrutsContentTypeRceDetector",
+    version = "0.1",
+    description = "Tsunami detector plugin for Apache Struts Command Injection via Content-Type header (CVE-2017-5638).",
```

Comment:
nit: max line width: 100. ditto for the remaining of this file and the unittest.

Reply:
Done